source/common/upstream/health_checker_impl.cc

```diff
@@ -488,9 +492,11 @@ void GrpcHealthCheckerImpl::GrpcActiveHealthCheckSession::onInterval() {
   request_encoder_ = &client_->newStream(*this);
   request_encoder_->getStream().addCallbacks(*this);
 
-  auto headers_message = Grpc::Common::prepareHeaders(
-      parent_.cluster_.info()->name(), parent_.service_method_.service()->full_name(),
-      parent_.service_method_.name(), absl::nullopt);
+  const std::string& authority =
+      parent_.authority_value_ ? parent_.authority_value_.value() : parent_.cluster_.info()->name();
```

Comment:
Probably it is better to check it explicitly: `parent_.authority_value_.has_value()` (I think this is the preferred style). WDYT?

Reply:
Awesome! Thanks!